index.js

```diff
@@ -1,5 +1,56 @@
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
+const Funnel = require('broccoli-funnel');
+const Remark = require('broccoli-lint-remark');
+const mergeTrees = require('broccoli-merge-trees');
+const VersionChecker = require('ember-cli-version-checker');
+const extensions = require('markdown-extensions');
+
 module.exports = {
-  name: 'ember-cli-remark'
+	name: 'ember-cli-remark',
+
+	init() {
+		this._super.init && this._super.init.apply(this, arguments);
+
+		const checker = new VersionChecker(this);
+
+		if (checker.for('ember-qunit', 'npm').exists() || checker.for('ember-cli-qunit', 'npm').exists()) {
+			this._testGenerator = 'qunit';
+		} else if (checker.for('ember-mocha', 'npm').exists() || checker.for('ember-cli-mocha', 'npm').exists()) {
+			this._testGenerator = 'mocha';
+		}
+	},
+
+	included(app) {
+		this._super.included.apply(this, arguments);
+		this._options = app.options.remark || {};
```

Comment:
I would get the options from 'ember-cli-remark' instead of 'remark' because Those are options for the addon no for the remark library, i mean, when you create remark you pass only quiet and testGenerator. Its similar to ember-cli-workbox, you habe "ember-cli-workbox" options and "workbox" options.